chain/evm/client.go

```diff
@@ -55,20 +59,24 @@ func (client *Client) Tx(ctx context.Context, txID pack.Bytes) (account.Tx, pack
 	if err != nil {
 		return nil, pack.NewU64(0), fmt.Errorf(fmt.Sprintf("fetching tx by hash '%v': %v", txID, err))
 	}
-	chainID := tx.ChainId()
-	if client.ChainID != nil {
-		if chainID == nil {
-			return nil, 0, fmt.Errorf("nil chain ID")
-		}
-		if chainID.Cmp(client.ChainID) != 0 {
-			return nil, 0, fmt.Errorf("invalid chain ID, expected = %v, got = %v", client.ChainID.String(), chainID.String())
+
+	// Check the chain id for replay-protected tx.
+	if tx.Protected() {
```

Comment:
Just curious why this is needed I thought all txs after the fork will have this right?